### ISSUE TITLE
"conversion from int to DATE is unsupported" when calling ResultSet.getDate() on a result containing a NULL literal

### ISSUE THREAD

**Issue by lukaseder** [Microsoft/mssql-jdbc#482]
## Driver version or jar name

`com.microsoft.sqlserver:mssql-jdbc:6.2.1.jre8`

## SQL Server version

> Microsoft SQL Server 2014 - 12.0.2000.8 (X64) 
	Feb 20 2014 20:04:26 
	Copyright (c) Microsoft Corporation
	Express Edition (64-bit) on Windows NT 6.3 <X64> (Build 14393: )

## Client operating system

Windows 10

## Java/JVM version

> java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

## Table schema

N/A

## Problem description

Calling `ResultSet.getDate()` should work on all `NULL` values, regardless of their type.

## Expected behavior and actual behavior

Expected:

```java
assertNull(rs.getDate(1));
```

I've tried the repro code on MySQL, H2, Oracle (including a `from dual` clause, of course), PostgreSQL, and it always works as expected.

Actual:

```
com.microsoft.sqlserver.jdbc.SQLServerException: The conversion from int to DATE is unsupported.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:227)
	at com.microsoft.sqlserver.jdbc.DataTypes.throwConversionError(DataTypes.java:1612)
	at com.microsoft.sqlserver.jdbc.ServerDTVImpl.getValue(dtv.java:3779)
	at com.microsoft.sqlserver.jdbc.DTV.getValue(dtv.java:286)
	at com.microsoft.sqlserver.jdbc.Column.getValue(Column.java:189)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.getValue(SQLServerResultSet.java:1925)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.getValue(SQLServerResultSet.java:1906)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.getDate(SQLServerResultSet.java:2057)
	at SQLServer.main(SQLServer.java:62)
```

## Repro code

```java
try (Statement s = c.createStatement();
     ResultSet rs = s.executeQuery("select null")) {

    while (rs.next())
        System.out.println(rs.getDate(1));
}
```

**Comment by pwilkin** [comment]
Will add a comment instead of adding a new issue - this is a general problem with manually specified NULL return values. Whenever there is a column that is manually specified as NULL in a query (`select NULL as COLUMNNAME`), it's always autocast to INT and then any conversion basically fails. I had to work around this (had exactly the same problem just now with an IMAGE column that I specify as NULL in my select for a lazy-loading scenario) by doing `select CAST(NULL as IMAGE)`.

**Comment by v-afrafi** [comment]
@lukaseder @pwilkin Thanks for pointing out this issue. Could you please try and test pr #488. That should resolve the issue. Thanks!

**Comment by lukaseder** [comment]
@v-afrafi Thanks for the quick fix. As far as I'm concerned, I'll trust you on your judgment and wait until the release of this issue - I'm not in a hurry :)

Thanks again

**Comment by lukaseder** [comment]
Hmm, any plans of tagging closed issues with milestones / release versions? Would be cool to see this kind of information here on github

**Comment by v-afrafi** [comment]
@lukaseder  This fix was made with 6.3.3.jre*-preview and will be included in the next production ready release (version 6.4.0) . Thanks for the idea, will definitely keep that information on the issues as well!
